### PR TITLE
[flutter_tools] fix multiple defines in flutter tooling, web

### LINKED ISSUE #54825: Multiple --dart-define cause crash during android apk build
When running
```
flutter build apk --dart-define=A=a --dart-define=B=b
```
the build failes with the message
```
Target kernel_snapshot failed: Exception: The value of -DDartDefines is not formatted correctly.
The value must be a JSON-encoded list of strings but was:               
["A=a"                                                                  
build failed. 
``` 

```
flutter build apk --dart-define=A=a
```
runs as expected.

Flutter doctor output:
```
[✓] Flutter (Channel beta, v1.17.0, on Mac OS X 10.14.6 18G3020, locale en-NL)
    • Flutter version 1.17.0 at /Users/david/Android/flutter
    • Framework revision d3ed9ec945 (8 days ago), 2020-04-06 14:07:34 -0700
    • Engine revision c9506cb8e9
    • Dart version 2.8.0 (build 2.8.0-dev.18.0 eea9717938)

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /Users/david/Library/Android/sdk
    • Platform android-29, build-tools 29.0.3
    • ANDROID_HOME = /Users/david/Library/Android/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_212-release-1586-b4-5784211)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.3.1, Build version 11C504
    • CocoaPods version 1.8.4

[!] Android Studio (version 3.6)
    • Android Studio at /Applications/Android Studio.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build
      1.8.0_212-release-1586-b4-5784211)

[✓] VS Code (version 1.44.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.9.1

[✓] Connected device (1 available)
    • iPhone 8 • AA328C41-79A7-448F-B5E7-B7C5D058F337 • ios •
      com.apple.CoreSimulator.SimRuntime.iOS-13-3 (simulator)

! Doctor found issues in 1 category.
```

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Multiple dart-defines could not be passed to the tool. A single define happened to work through a quirk in the formatting. The issue is that package:args would seemingly parse the value incorrectly in a way that happened to work for a single value, but broke if multiple values were provided.

Update the tooling to use a similar strategy to the extra front end options, Update the formatting to use a comma-separated list.

Updates the dart-define tests to pass two values.

Adds a dart-define test for web debug and release mode. Fixes a bug in web release mode define handling

Fixes https://github.com/flutter/flutter/issues/54825

### CODE REVIEW

**Comment by jonahwilliams** [comment]
This is why you always write tests with zero, one, and many inputs

File: packages/flutter_tools/lib/src/commands/assemble.dart [hunk 1]
```diff
@@ -182,12 +183,17 @@ class AssembleCommand extends FlutterCommand {
...
+    if (argResults.wasParsed(kDartDefines)) {
+      results[kDartDefines] = argResults[kDartDefines] as String;
+    }
     return results;
   }
 
   @override
   Future<FlutterCommandResult> runCommand() async {
     final List<Target> targets = createTargets();
+    print(createEnvironment().defines);
```
Comment (jmagman) [inline packages/flutter_tools/lib/src/commands/assemble.dart:196]: Leftover?

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/commands/assemble.dart]: Fixed

**Review by jmagman — APPROVED** [review]
LGTM

**Comment by jonahwilliams** [comment]
None of the tests ran 🤔

**Pull request merged**